### PR TITLE
feat: saving stats for SmartLayer

### PR DESCRIPTION
**Pull request by sonic182**

### CODE REVIEW

File: playground/demo/assets/smart/events.js [hunk 1]
```diff
@@ -76,20 +102,184 @@ export class Events {
     scrollToTop();
   }
 
-
   _onScroll(ev, scroll_top_element) {
     let smart = this.smart;
 
+    // TODO: remove this. It isn't needed if we are using the
```
Comment (resetreboot) [inline playground/demo/assets/smart/events.js]: We are using version control, remove commented code.

File: playground/demo/assets/smart/events.js [hunk 2]
```diff
@@ -76,20 +102,184 @@ export class Events {
...
+          this.scroll_timeout = null
+        }
+
+        this.scroll_timeout = setTimeout(() => {
+          this.stats_handler.registerScrollDown(scroll_top_element.scrollTop)
+          this.scroll_timeout = null
+        }, 1000)
+
+        smart.elements.arrow.display = true
+      } else {
```
Comment (resetreboot) [inline playground/demo/assets/smart/events.js]: Maybe check if we have to stop the event we start before? We may hide the arrow but the event will be fired.

File: playground/demo/assets/smart/events.js [hunk 3]
```diff
@@ -76,20 +102,184 @@ export class Events {
...
+    for (let result of last_response.results){
+      pos += 1;
+      if (result.dfid == product.dfid)
+        break;
+    }
+    // increase page position
+    // if page is 1, the increase will be 0
+    pos += last_response.results_per_page * (last_response.page - 1)
+
+    this.stats.registerEvent('smart_autofacets', {
```
Comment (resetreboot) [inline playground/demo/assets/smart/events.js]: Shouldn't this event be `smart_click` or something similar?

File: playground/demo/assets/smart/events.js [hunk 4]
```diff
@@ -76,20 +102,184 @@ export class Events {
...
+  _isAutoFacet(facet) {
+    const f = this.autofacets.filter(fct => {
+      return (fct.field === facet.field && 
+        fct.value === fct.value);
+    });
+
+    return f.length > 0;
+  }
+
+  // Put the newly activated facets into our own
```
Comment (resetreboot) [inline playground/demo/assets/smart/events.js]: Remove commented code, I don't think we'll ever look back on that code.

File: playground/demo/assets/smart/events.js (hunk 3 above)
Comment (sonic182) [reply playground/demo/assets/smart/events.js]: Oh yes! I forgot to change it here

File: playground/demo/assets/smart/events.js (hunk 4 above)
Comment (sonic182) [reply playground/demo/assets/smart/events.js]: Done

**Pull request merged**